runtime/type-variable.js

```diff
@@ -45,8 +67,49 @@ class TypeVariable {
     }
   }
 
+  maybeMergeConstraints(variable) {
```

Comment:
Could be worth adding brief documentation for this and similar methods below.

Reply:
Done